lib/universal_peripherial.dart

```diff
@@ -111,55 +115,61 @@ class _ExpectAck extends _State {
 
 class _Idle extends _State {}
 
-class _FeatureExchange extends _State {
-  late Iterator<String> currentFeature;
+class _IterableExchange extends _State {
+  late List<String> result;
+  late Iterator<String> iter;
+  late String key;
+  late _State next;
+
+  _IterableExchange(this.iter, this.key, this.next, this.result) {}
 
   @override
   void onEnter() {
-    currentFeature = this._context.features.keys.iterator;
-    moveToNextFeature();
+    moveToNext();
   }
 
-  void moveToNextFeature() {
-    if (!currentFeature.moveNext())
-      this._context.transitionTo(_Idle());
+  void moveToNext() {
+    if (!iter.moveNext())
+      this._context.transitionTo(next);
     else
-      this._context.send("feature " + currentFeature.current);
+      this._context.send("$key ${iter.current}");
   }
 
   @override
   void onPeripheralCmd(String cmd) {
     if (cmd == 'ok')
-      this._context.features[currentFeature.current] = true;
+      this.result.add(iter.current);
     else if (cmd == 'nok')
-      this._context.features[currentFeature.current] = false;
+      print("${iter.current} not supported by peripherial");
     else
       super.onPeripheralCmd(cmd);
 
-    moveToNextFeature();
+    moveToNext();
   }
 }
 
 class _SyncVariant extends _State {
```

Comment:
Maybe `SyncVariantState` instead `_SyncVariant`?

Reply:
And `_State` to `PeripheralState`?

Reply:
I'm not sure. Implementation is internal and this is why I added `_`. Do we really want to add State for all states here?

Reply:
> I'm not sure. Implementation is internal and this is why I added `_`. 

Hmm, I got that idea. And as I see it uses in dart normally. Just it very hart to read for me personally. Thinking if here is another solution..

> Do we really want to add State for all states here?

Yes. Will be more readable lines like `transitionTo(..State)`